src/Traits/MockTrait.php

```diff
@@ -129,30 +130,63 @@ protected function assertArguments(
         string $class,
         string $function,
         int $callIndex,
-        bool $isClass = true
+        bool $isClass = true,
     ): void {
+        $reflection = $isClass
+            ? new ReflectionMethod($class, $function)
+            : new ReflectionFunction($function);
+
         $expectedCount = count($expected);
         $actualCount = count($actual);
+        $parameters = $reflection->getParameters();
+        $requiredParametersCount = count(array_filter($parameters, fn ($param) => !$param->isOptional()));
+
+        $this->assertArgumentCount($expectedCount, $actualCount, $requiredParametersCount, $function);
```

Comment:
let's move counting into the assertArgumentCount function

```suggestion
        $this->assertArgumentCount($expected, $actual, $reflectionArgs, $function);
```